src/dask_awkward/pickle.py

```diff
@@ -0,0 +1,66 @@
+from __future__ import annotations
+
+__all__ = ("plugin",)
+
+import pickle
+
+import awkward as ak
+
+
+def pickle_record(record: ak.Record, protocol: int) -> tuple:
+    layout = ak.to_layout(record, allow_record=True)
+    form, length, container = ak.operations.to_buffers(
+        layout.array,
+        buffer_key="{form_key}-{attribute}",
+        form_key="node{id}",
+        byteorder="<",
+    )
+
+    # For pickle >= 5, we can avoid copying the buffers
+    if protocol >= 5:
+        container = {k: pickle.PickleBuffer(v) for k, v in container.items()}
+
+    if record.behavior is ak.behavior:
+        behavior = None
+    else:
+        behavior = record.behavior
+
+    return (
+        object.__new__,
+        (ak.Record,),
+        (form.to_dict(), length, container, behavior, layout.at),
+    )
+
+
+def pickle_array(array: ak.Array, protocol: int) -> tuple:
+    layout = ak.to_layout(array, allow_record=True)
+    form, length, container = ak.operations.to_buffers(
+        layout,
+        buffer_key="{form_key}-{attribute}",
+        form_key="node{id}",
+        byteorder="<",
+    )
+
+    # For pickle >= 5, we can avoid copying the buffers
+    if protocol >= 5:
+        container = {k: pickle.PickleBuffer(v) for k, v in container.items()}
+
+    if array.behavior is ak.behavior:
+        behavior = None
+    else:
+        behavior = array.behavior
+
+    return (
+        object.__new__,
+        (ak.Array,),
+        (form.to_dict(), length, container, behavior),
+    )
```

Comment:
For now this is just `to_buffers` with _no_ packing. In future we can implement a form-preserving pack operation, perhaps in Awkward Array

Reply:
No packing is not a bad way to start for serialization in Dask (as long as it's not the default for pickle outside of Dask).